Sources/MapboxNavigation/DestinationPreviewViewController.swift

```diff
@@ -0,0 +1,126 @@
+import UIKit
+import CoreLocation
+import MapboxDirections
+
+// :nodoc:
+public class DestinationPreviewViewController: DestinationPreviewing {
```

Comment:
It seems like this view controller previews a whole route, not just one of its destinations. The CarPlay support code says “preview” throughout, so it’s probably going to be less confusing down the road if we call it just previewing instead of destination previewing. That said, “Previewing” would be a very generic protocol name. How about something like “WaypointPresenting”?

Reply:
Current naming attempts to also mimic Android navigation states. For example, here's what they have:
- FreeDrive
- DestinationPreview
- RoutePreview
- ActiveNavigation
- Arrival

`WaypointPresenting`, sounds reasonable, but we'll have to rename `DestinationOptions` as well, and naming like `WaypointOptions`, `PresentedWaypointOptions` doesn't seen to fully describe its functionality. I'll probably keep naming the way it is and re-consider it after landing `PreviewViewController` itself.